File: BOJ/Greedy/1931/1931.cpp
// 1931 회의실 배정
// https://www.acmicpc.net/problem/1931
#include <iostream>
#include <vector>
#include <algorithm>

using namespace std;

int N;
vector<pair<int,int>> rooms;

bool compare(pair<int, int> a, pair<int, int> b) {
    if (a.second == b.second) return a.first < b.first;
    return a.second < b.second;
}

int solution(void)
{
    int answer = 1;

    sort(rooms.begin(), rooms.end(), compare);
    int cmpEndTime = rooms[0].second;

    for (int i = 1; i < rooms.size(); ++i)
    {
        if (cmpEndTime <= rooms[i].first) 
        {
            cmpEndTime = rooms[i].second;
            answer++;
        }
    }

    return answer;
}

int main(void) {
    cin >> N;
    int s, e;

    for (int i = 0; i < N; ++i)
    {
        cin >> s >> e;
        rooms.push_back({s,e});
    }

    cout << solution();
    return 0;
}

File: BOJ/Greedy/1931/1931.py
# 1931 회의실 배정
# https://www.acmicpc.net/problem/1931

import sys
input = sys.stdin.readline


def solution(n, info):
    ret = 0

    info.sort(key=lambda x: (x[1], x[0]))
    currentEnding = info[0][1]
    ret += 1

    for i in range(1, n):
        if info[i][0] >= currentEnding:
            currentEnding = info[i][1]
            ret += 1

    return ret


n = int(input())
info = [list(map(int, input().split())) for _ in range(n)]

print(solution(n, info))
